tests/goth_tests/test_run_custom_usage_counter.py

```diff
@@ -0,0 +1,103 @@
+"""An integration test scenario that runs custom usage counter example requestor app."""
+import logging
+import os
+from pathlib import Path
+import re
+import time
+from typing import List
+
+import pytest
+
+from goth.assertions import EventStream
+from goth.configuration import load_yaml, Override
+from goth.runner.log import configure_logging
+from goth.runner import Runner
+from goth.runner.probe import RequestorProbe
+
+from .assertions import assert_no_errors, assert_all_invoices_accepted
+
+
+logger = logging.getLogger("goth.test.run_custom_usage_counter")
+
+RUNNING_TIME = 200  # in seconds
+SUBNET_TAG = "goth"
+
+
+async def assert_counter_not_decremented(output_lines: EventStream[str]):
+    """Assert that the custom usage counter is not decremented."""
+    last_value = -1
+
+    async for line in output_lines:
+        m = re.search("total cost so far .* 'golem.usage.custom.counter': '([^']*)'", line)
+        if m:
+            value = float(m.group(1))
+            logger.info(f"Custom usage counter value: {value}")
+            if value < last_value:
+                raise AssertionError(f"Current custom usage counter was decremented.")
+            last_value = value
+
+
+async def assert_correct_startup_and_shutdown(output_lines: EventStream[str]):
+    """Assert that all providers and services that started stopped successfully."""
+    providers_names = set()
+    service_ids = set()
```

Comment:
This might be nitpicking, but why plural `providers` and singular `service`?

Reply:
It's not nitpicking. Fixed.